corehq/apps/sms/tasks.py

```diff
@@ -84,12 +84,24 @@ def remove_from_queue(queued_sms):
 
 def get_sms_from_queued_sms(queued_sms):
     sms = SMS()
-    for field in sms._meta.get_fields():
-        if field.name != 'id':
-            setattr(sms, field.name, getattr(queued_sms, field.name))
+    for field in _get_sms_fields_to_copy():
+        setattr(sms, field, getattr(queued_sms, field))
     return sms
 
 
+def _get_sms_fields_to_copy():
+    """Returns a set of field attribute names to copy from QueuedSMS to SMS.
+    This should be all fields in QueuedSMS except 'id'
+    """
+    res = set()
+    for field in QueuedSMS._meta.get_fields():
+        try:
+            res.add(field.attname)  # use attname to avoid DB lookups for related models
```

Comment:
Hey @snopoke ,

can you expand on this? How is this different now.

Reply:
in this case it's `messaging_subevent_id` instead of `messaging_subevent`.